tests/Aspire.NATS.Net.Tests/AspireNatsClientExtensionsTests.cs

```diff
@@ -97,8 +112,108 @@ public void ConnectionNameWinsOverConfigSection(bool useKeyed)
             host.Services.GetRequiredKeyedService<INatsConnection>("nats") :
             host.Services.GetRequiredService<INatsConnection>();
 
-        Assert.Equal(ConnectionString, connection.Opts.Url);
+        Assert.Equal(_connectionString, connection.Opts.Url);
         // the connection string from config should not be used since it was found in ConnectionStrings
         Assert.DoesNotContain("unused", connection.Opts.Url);
     }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public async Task AddNatsClient_HealthCheckShouldBeRegisteredWhenEnabled(bool useKeyed)
+    {
+        var key = DefaultConnectionName;
+        var builder = CreateBuilder(_connectionString);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedNatsClient(key, settings =>
+            {
+                settings.HealthChecks = true;
+            });
+        }
+        else
+        {
+            builder.AddNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = true;
+            });
+        }
+
+        var host = builder.Build();
+
+        var healthCheckService = host.Services.GetRequiredService<HealthCheckService>();
+
+        var healthCheckReport = await healthCheckService.CheckHealthAsync();
+
+        var healthCheckName = useKeyed ? $"NATS_{key}" : "NATS";
+
+        Assert.Contains(healthCheckReport.Entries, x => x.Key == healthCheckName);
+    }
+
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void AddNatsClient_HealthCheckShouldNotBeRegisteredWhenDisabled(bool useKeyed)
+    {
+        var builder = CreateBuilder(_connectionString);
+
+        if (useKeyed)
+        {
+            builder.AddKeyedNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = false;
+            });
+        }
+        else
+        {
+            builder.AddNatsClient(DefaultConnectionName, settings =>
+            {
+                settings.HealthChecks = false;
+            });
+        }
+
+        var host = builder.Build();
+
+        var healthCheckService = host.Services.GetService<HealthCheckService>();
+
+        Assert.Null(healthCheckService);
+    }
+
+    [RequiresDockerFact]
+    public async Task NatsInstrumentationEndToEnd()
+    {
+        var builder = CreateBuilder(_connectionString);
+
+        builder.AddNatsClient(DefaultConnectionName, settings =>
+        {
+            settings.Tracing = true;
+        });
+
+        var notifier = new ActivityNotifier();
```

Comment:
Since we are adding an ActivityListener, we should be splitting these tests to run out of process using RemoteExecutor. The reason is because ActivityListeners are global for the whole process. So if another test is running at the same time and emits an Activity, this listener is going to be notified.

See the test changes in https://github.com/dotnet/aspire/pull/2982/files#diff-50442307fcbb0ffc7a9ac85f2dc7cbc0d51b0f666d2e9f74df47713d72f8196bR259 for how to do this.